cmd/kubeadm/app/phases/certs/certlist.go

```diff
@@ -204,6 +207,22 @@ func (c Certificates) AsMap() CertificateMap {
 	return certMap
 }
 
+var errInvalid = errors.New("invalid argument")
+
+type certificatesVisitor func(*KubeadmCert) error
+
+func (c Certificates) visit(visitor certificatesVisitor) error {
+	if visitor == nil {
+		return fmt.Errorf("%w: visitor was nil", errInvalid)
+	}
+	for _, cert := range c {
+		if err := visitor(cert); err != nil {
+			return err
+		}
+	}
+	return nil
+}
```

Comment:
This allows me to unit test the  behaviour of Certificates list iteration with simple visitor functions.
Avoids having to have slow unit tests that create lists of keys and CSR files.